components/MediList/index.js

```diff
@@ -2,10 +2,10 @@ import styled from 'styled-components';
 import { useState } from 'react';
 import { dummyMedications } from '../../dummydata';
 
-
 export default function MediList() {
   const [medikament, setMedikament] = useState('');
   const [dosage, setDosage] = useState('');
```

Comment:
Why do you have a second state for your dosage? As far as I understand the dosage is always specified for a medication. You could of course relate it with a foreign key (like with relational databases) everytime. But I think in this use case it might be easier to store objects in an array:

const [medication, setMedication] = useState(dummyMedications)

If you really need that controlled input you can use a second state with the same data structure just for the currentMedication (object). 

const [currentMedication, setCurrentMedication] = useState({medicationName: "", dosage: "";})